python/cuml/test/test_fil.py

```diff
@@ -0,0 +1,285 @@
+from sklearn.metrics import accuracy_score, mean_squared_error
+from sklearn.model_selection import train_test_split
+
+if has_xgboost():
+    import xgboost as xgb
+
+    def simulate_data(m, n, k=2, random_state=None, classification=True):
+        if classification:
+            features, labels = make_classification(n_samples=m,
+                                                   n_features=n,
+                                                   n_informative=int(n/5),
+                                                   n_classes=k,
+                                                   random_state=random_state)
+        else:
+            features, labels = make_regression(n_samples=m,
+                                               n_features=n,
+                                               n_informative=int(n/5),
+                                               n_targets=1,
+                                               random_state=random_state)
+        return np.c_[features].astype(np.float32), \
+            np.c_[labels].astype(np.float32)
+
+
+def unit_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.unit)
+
+
+def quality_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.quality)
+
+
+def stress_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.stress)
+
+
+def _build_and_save_xgboost(model_path,
+                            X_train,
+                            y_train,
+                            classification=True,
+                            num_rounds=5,
+                            xgboost_params={}):
+    """Trains a small xgboost classifier and saves it to model_path"""
+    dtrain = xgb.DMatrix(X_train, label=y_train)
+
+    # instantiate params
+    params = {'silent': 1}
+
+    # learning task params
+    if classification:
+        params['eval_metric'] = 'error'
+        params['objective'] = 'binary:logistic'
+    else:
+        params['eval_metric'] = 'error'
+        params['objective'] = 'reg:squarederror'
+        params['base_score'] = 0.0
+
+    params.update(xgboost_params)
+
+    bst = xgb.train(params, dtrain, num_rounds)
+    bst.save_model(model_path)
+    return bst
+
+
+@pytest.mark.parametrize('n_rows', [unit_param(100),
+                                    quality_param(1000),
+                                    stress_param(500000)])
+@pytest.mark.parametrize('n_columns', [unit_param(11),
+                                       quality_param(100),
+                         stress_param(1000)])
+@pytest.mark.parametrize('num_rounds', [unit_param(1),
+                                        unit_param(5),
+                                        quality_param(50),
+                                        stress_param(90)])
+@pytest.mark.skipif(has_xgboost() is False, reason="need to install xgboost")
+def test_fil_class(n_rows, n_columns, num_rounds, tmp_path):
```

Comment:
`class` sounds ambiguous here (class or classification?), `test_fil_classification` or `test_fil_classify` would be better.